File: Apps/gm-web-app/src/app/stocks/stock-data.ts
export class StockData {
    constructor(public date: Date,
        public symbol: string,
        public open: number,
        public high: number,
        public low: number,
        public close: number,
        public adjustedClose: number,
        public volume: number,
        public previousClose: number) {

    }

    public get percentChangeFromPreviousClose(): number {
        if(this.previousClose == 0.0) {
            return 0.0;
        }
        return 100.0 * (this.close - this.previousClose) / this.previousClose;
    }
}

File: Apps/gm-flutter-app/gimmillions/lib/views/stocks/stock-data.dart
class StockData {
  DateTime date;
  String symbol;
  double open;
  double high;
  double low;
  double close;
  double adjustedClose;
  double volume;
  double previousClose;

  StockData(this.date, this.symbol, this.open, this.high, this.low, this.close,
      this.adjustedClose, this.volume, this.previousClose);
}
